### PR TITLE
[BUG FIX]: Fix unecessary minus 1 for OpenCL function in surface width/height/pitch

### PR DESCRIPTION
**Pull request by quintinwang5**

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp [hunk 1]
```diff
@@ -140,19 +140,10 @@ class LoadStorePrefetchOpConversion
...
-      if (multiplyBytes)
-        truncatedShape = mul(truncatedShape, bytes);
-      return sub(truncatedShape, i32_val(1));
-    };
-
-    Value surfaceW = calculateSurface(ptrOp.getShape()[1], true);
-    Value surfaceH = calculateSurface(ptrOp.getShape()[0], false);
-    Value surfaceP = calculateSurface(ptrOp.getStrides()[0], true);
+    Value surfaceW = ptrOp.getShape()[1];
+    Value surfaceH = ptrOp.getShape()[0];
```
Comment (Dewei-Wang-sh) [inline third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp:145]: we may still need trunc

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 2]
```diff
@@ -265,9 +265,18 @@ createGenISA2DBlockRead(TritonGEN::Matrix2DBlockLoadOp op,
 
   auto moduleOp = rewriter.getBlock()->getParent()->getParentOfType<ModuleOp>();
   Value ptr = op.getPtr();
-  Value baseWidth = op.getBaseWidth();
-  Value baseHeight = op.getBaseHeight();
-  Value basePitch = op.getBasePitch();
+  Value elemSizeInBytes = i32_val(op.getElemSizeInBits() / 8);
+
+  auto calculateSurface = [&](Value shape, bool multiplyBytes) {
```
Comment (Dewei-Wang-sh) [inline third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:270]: we should extract this as a function and reused

**Review by Dewei-Wang-sh — APPROVED** [review]
lgtm

**Comment by etiotto** [comment]
@quintinwang5 the test are failing, Just a reminder.

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 3]
```diff
@@ -334,9 +353,15 @@ createGenISA2DBlockWrite(TritonGEN::Matrix2DBlockStoreOp op,
   Location loc = op->getLoc();
 
   Value ptr = op.getPtr();
-  Value baseWidth = op.getBaseWidth();
-  Value baseHeight = op.getBaseHeight();
-  Value basePitch = op.getBasePitch();
+  Value elemSizeInBytes = i32_val(op.getElemSizeInBits() / 8);
```
Comment (quintinwang5) [inline third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:356]: @whitneywhtsang  Store here.

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 4]
```diff
@@ -405,9 +430,15 @@ createGenISA2DBlockPrefetch(TritonGEN::Matrix2DBlockPrefetchOp op,
...
```
Comment (quintinwang5) [inline third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:433]: Prefetch here.

**Pull request merged**